src/control/server/instance_drpc.go

```diff
@@ -75,3 +79,92 @@ func (srv *IOServerInstance) CallDrpc(module, method int32, body proto.Message)
 
 	return makeDrpcCall(dc, module, method, body)
 }
+
+// drespToMemberResult converts drpc.Response to system.MemberResult.
+//
+// MemberResult is populated with rank, state and error dependent on processing
+// dRPC response. Target state param is populated on success, Errored otherwise.
+func drespToMemberResult(rank system.Rank, action string, dresp *drpc.Response, err error, tState system.MemberState) *system.MemberResult {
+	var outErr error
+	state := system.MemberStateErrored
+
+	if err != nil {
+		outErr = errors.WithMessagef(err, "rank %s dRPC failed", &rank)
+	} else {
+		resp := &mgmtpb.DaosResp{}
+		if err = proto.Unmarshal(dresp.Body, resp); err != nil {
+			outErr = errors.WithMessagef(err, "rank %s dRPC unmarshal failed",
+				&rank)
+		} else if resp.GetStatus() != 0 {
+			outErr = errors.Errorf("rank %s dRPC returned DER %d",
+				&rank, resp.GetStatus())
```

Comment:
Are we able to translate the numeric DER value to a string? I thought we were doing that elsewhere in the control plane.

Reply:
yes will look into that as part of follow one reformat PR